### PR TITLE
Fix Frontend Tests

### PR DESCRIPTION
**Pull request by galabastro**
```
fix: update npm lock file
fix: missing set user id functions
fix: make frontend tests into pytests
```

🚀 Pull Request
========================

🎯 PR Type
----------
-   [ ] **Feature:** Adds new functionality or user-facing change.
-   [x] **Bug Fix:** Solves a defect or incorrect behavior.
-   [ ] **Refactor:** Improves code structure/readability without changing external behavior.
-   [ ] **Documentation:** Updates to README, Wiki, or internal docs.
-   [ ] **Chore:** Non-code changes (e.g., CI configuration, dependency bumps).

1\. Issue Tracking & Reference
------------------------------
-   **Ticket Number(s):** N/A
-   **Ticket Link(s):** N/A

2\. Summary of Changes
----------------------
Fixes all failing frontend tests in CI. The GitHub Actions `frontend-tests` job was failing due to an out-of-sync `package-lock.json`, three failing vitest unit tests, and two Python test files crashing pytest collection with `SystemExit`.

-   **Expected Outcome:** All frontend-tests and python-tests CI jobs pass cleanly.

### Detailed Change Log & Justification
-   **Change 1:** Updated `frontend_service/package-lock.json` to sync with `package.json`.
    -   **Justification:** `package.json` required `@newrelic/browser-agent@^1.312.1` but the lock file still referenced `1.296.0`, causing `npm ci` to fail with `EUSAGE`. Running `npm install` resolved the version and added the missing `@newrelic/rrweb*` transitive packages.
-   **Change 2:** Added `setUserId`, `userId`, `browserUserId`, and `setBrowserUserId` to `mockLoginContext` in `app/routes/login.test.tsx`.
    -   **Justification:** `login.tsx` destructures `setUserId` from `LoginContext` and calls it on successful login. The mock was missing this field, causing `setUserId is not a function` at runtime and the `submits login form with credentials` test to fail.
-   **Change 3:** Added `userId` to `mockLoginContext` and prepended an `/accounts-service/users/by-id/` fetch mock to every test in `app/components/payments/PayBillCard.test.tsx`.
    -   **Justification:** `PayBillCard.tsx` guards `fetchPaymentMethods` with `if (!userId) { return; }`. With no `userId` in the mock, the component never called fetch, so the two tests asserting `/bill-pay-service/payment-methods/` was called always failed. The component also calls the accounts-service first to retrieve `stripe_customer_id`, so each test needed that mock prepended.
-   **Change 4:** Converted `tests/test_mfe_single.py` from a standalone script to a proper pytest file.
    -   **Justification:** The file called `exit(1)` at module level when `NEW_RELIC_USER_API_KEY` was missing, causing pytest to crash with `INTERNALERROR> SystemExit: 1` during collection and marking the file as failed even though no tests ran.
-   **Change 5:** Converted `tests/test_with_interactions.py` from a standalone script to a proper pytest file with two test functions.
    -   **Justification:** Same `exit(1)` at module level issue as above. Additionally, the file had a hardcoded `localhost:3000` URL replaced with the `RELIBANK_URL` environment variable, and the implicit `if count > 0` verification block was split into two explicit asserting test functions.

3\. Testing
-----------

### Testing Strategy
-   **Environment Used:** Local and GitHub Actions (CI)
-   **Production Safety Assessment:** All changes are confined to test files and the `package-lock.json`. No application source code was modified.
-   **What areas were NOT tested and why?** The `test_mfe_single.py` and `test_with_interactions.py` tests require `NEW_RELIC_USER_API_KEY` and a live environment to fully execute; they were verified to collect and skip cleanly without the key.

### Coverage Details

| Scenario | Details | Results |
| --- | --- | --- |
| Happy Path | All 37 vitest tests pass locally | ✅ 37 passed |
| Happy Path | `test_mfe_single.py` and `test_with_interactions.py` collect and skip cleanly without API key | ✅ 3 skipped |
| Edge Case 1 | `PayBillCard` tests with `userId` set — accounts-service mock consumed before payment-methods mock | ✅ All 6 PayBillCard tests pass |
| Edge Case 2 | `login.test.tsx` — `setUserId` called after successful auth response | ✅ All 4 login tests pass |

4\. Evidence
------------
-   **Screenshots/Gifs:** N/A
-   **Console/Log Output Snippets:**
```
 Test Files  5 passed (5)
      Tests  37 passed (37)
   Start at  11:23:38
   Duration  7.21s
```
```
test_mfe_single.py::test_mfe_timing_events_exist SKIPPED (NEW_RELIC_...)
test_with_interactions.py::test_mfe_timing_events_exist SKIPPED (NEW_...)
test_with_interactions.py::test_mfe_timing_event_details SKIPPED (NE_...)
3 skipped in 0.07s
```
-   **GitHub Actions:** `frontend-tests` job passed on CI run #126 (all 37 vitest tests green, no `npm ci` errors).
-   **Test Report Links:** N/A

_Testing and evidence are **REQUIRED** before requesting review!_

5\. Review and Merge Checklist
------------------------------
-   [x] All blocking review comments from the latest round are **resolved**.
-   [x] This branch has been rebased against the `main` branch and all **merge conflicts are resolved**.
-   [x] The code includes sufficient comments, particularly in complex or non-obvious areas.
-   [x] The code adheres to project **coding standards** (linting, style guides, best practices).
-   [x] Relevant documentation and runbooks have been updated, if necessary.
-   [x] My changes generate no new warnings or errors.
-   [x] My commits are squashed into a single, descriptive commit.
-   [x] My test results are uploaded as a text file, and new tests were created where required.

### CODE REVIEW

**Comment by galabastro** [comment]
successful test run: https://github.com/newrelic/relibank/actions/runs/25071352767

File: tests/test_mfe_single.py [hunk 1]
```diff
@@ -1,36 +1,73 @@
...
+    nrql = f"""
+        SELECT count(*) FROM MicroFrontEndTiming
+        WHERE source.id IN ({MFE_SOURCE_IDS})
+        SINCE 15 minutes ago
+    """
+
+    print(f"Querying NRQL: {nrql}")
+    results = query_nrql(nrql)
+    count = results[0]["count"] if results else 0
+    print(f"MicroFrontEndTiming events found: {count}")
```
Comment (thezackm) [inline tests/test_mfe_single.py]: this NRQL will give you a false positive if any one of the MFEs shows traffic. You should `FACET source.id` and then ensure each facet is `> 0` to make sure each one is reporting as expected.

File: tests/test_with_interactions.py [hunk 2]
```diff
@@ -1,54 +1,99 @@
...
+    nrql = f"""
+        SELECT count(*) FROM MicroFrontEndTiming
+        WHERE source.id IN ({MFE_SOURCE_IDS})
+        SINCE 15 minutes ago
+    """
+
+    print(f"Querying NRQL: {nrql}")
+    results = query_nrql(nrql)
+    count = results[0]["count"] if results else 0
+    print(f"MicroFrontEndTiming events found: {count}")
```
Comment (thezackm) [inline tests/test_with_interactions.py]: same comment as the other test; this needs a `FACET` to ensure that each MFE is functional

File: tests/test_mfe_single.py [hunk 3]
```diff
@@ -1,36 +1,73 @@
...
-if not NEW_RELIC_API_KEY:
-    print("ERROR: NEW_RELIC_USER_API_KEY environment variable not set")
-    print("Usage: export NEW_RELIC_USER_API_KEY='your-key' && python test_mfe_single.py")
-    exit(1)
+MFE_SOURCE_IDS = ", ".join([
+    "'550e8400-e29b-41d4-a716-446655440000'",
+    "'650e8400-e29b-41d4-a716-446655440001'",
+    "'750e8400-e29b-41d4-a716-446655440002'",
+    "'850e8400-e29b-41d4-a716-446655440003'",
+])
```
Comment (thezackm) [inline tests/test_mfe_single.py]: why are these statically-defined? are we positive they'll never change?

File: tests/test_with_interactions.py [hunk 4]
```diff
@@ -1,54 +1,99 @@
...
-if not NEW_RELIC_API_KEY:
-    print("ERROR: NEW_RELIC_USER_API_KEY environment variable not set")
-    print("Usage: export NEW_RELIC_USER_API_KEY='your-key' && python test_with_interactions.py")
-    exit(1)
+MFE_SOURCE_IDS = ", ".join([
+    "'550e8400-e29b-41d4-a716-446655440000'",
+    "'650e8400-e29b-41d4-a716-446655440001'",
+    "'750e8400-e29b-41d4-a716-446655440002'",
+    "'850e8400-e29b-41d4-a716-446655440003'",
+])
```
Comment (thezackm) [inline tests/test_with_interactions.py]: same question as above re: static values here

**Review by thezackm — CHANGES_REQUESTED** [review]

File: tests/test_mfe_single.py (hunk 3 above)
Comment (galabastro) [reply tests/test_mfe_single.py]: https://docs-preview.newrelic.com/docs/browser/register#parameters

it's a required id that we get to define:
```
id (string or number): Unique identifier for the micro frontend. The ID supplied should be generated and managed according to your organization's internal processes and conventions. It serves as a permanent, stable reference to uniquely identify this specific entity across your system. Ensure this value remains consistent and is not duplicated across other entities.
```

the ids are something that are defined by us, and whenever we get new environments, it's something we miiight have to vary if the api key doesn't disambiguate the environment

File: tests/test_mfe_single.py (hunk 1 above)
Comment (galabastro) [reply tests/test_mfe_single.py]: this is a great callout, i'll update the nrql

File: tests/test_with_interactions.py (hunk 4 above)
Comment (galabastro) [reply tests/test_with_interactions.py]: see: https://github.com/newrelic/relibank/pull/77#discussion_r3169346573

File: tests/test_mfe_single.py (hunk 1 above)
Comment (galabastro) [reply tests/test_mfe_single.py]: alright! the new tests are testing for each microfrontend component:
```
MicroFrontEndTiming event counts by source.id:                                                                                                            
  550e8400-e29b-41d4-a716-446655440000: 2  (AdBanner)                                                                                                     
  650e8400-e29b-41d4-a716-446655440001: 2  (SpendingChart)                                                                                                
  750e8400-e29b-41d4-a716-446655440002: 2  (SpendingCategories)                                                                                           
  850e8400-e29b-41d4-a716-446655440003: 2  (AccountBalanceTrends)                                                                                         
✅ All 4 MFEs reported MicroFrontEndTiming events                                                                                                         
PASSED (3:35)  
```

File: tests/test_with_interactions.py (hunk 2 above)
Comment (galabastro) [reply tests/test_with_interactions.py]: see: https://github.com/newrelic/relibank/pull/77#discussion_r3169763469

**Review by thezackm — APPROVED** [review]
LGTM 🚀
